Updating samples to v1 version

soltysh opened the pull request:
I'm currently updating all the json samples to work with v1 version of our API.

// cc @bparees @mfojtik @danmcp @adellape

bparees commented on dev_guide/integrating_external_services.adoc:
```diff
@@ -53,12 +53,20 @@ ignore the service and allow you to specify endpoints manually:
   "metadata": {
     "name": "example-mysql-service"
   },
-  "spec": { <1>
```
you removed the link to the footnote.  though i'm not sure what that footnote is trying to say anyway.

soltysh replied:
I think there should be added:
```
        "selector": {
        }
```
to which the footnote should be referring to, but I'm not sure if it makes sense. @mfojtik your knowledge is needed in this topic, would adding empty selector match what the footnote is saying?

bparees replied:
yeah that's what it was supposed to link to and if you read the paragraph above (which says "leave the selector field unset") then it makes sense i guess.

soltysh replied:
Yeah, I'm not an expert in that area, but @mfojtik was dealing with them, so I'm hoping to get his answer on this. Clearly that was an error in the docs.

bparees commented on dev_guide/integrating_external_services.adoc:
```diff
@@ -77,11 +85,11 @@ the location to send traffic going to the service.
-  "apiVersion": "v1beta3",
+  "apiVersion": "v1",
   "metadata": {
-    "name": "external-mysql-service" <1>
```
why did you remove the footnote link?

bparees commented on dev_guide/integrating_external_services.adoc:
```diff
@@ -77,11 +85,11 @@ the location to send traffic going to the service.
-    "name": "external-mysql-service" <1>
+    "name": "external-mysql-service"
   },
-  "endpoints": [ <2>
```
again

bparees commented on dev_guide/integrating_external_services.adoc:
```diff
@@ -100,63 +108,70 @@ service by setting environment variables in the appropriate containers. In the f
   "metadata": {
     "name": "my-app-deployment"
   },
-  "template": { <1>
```
footnote

adellape commented:
Wowee, :bow:.

soltysh commented:
@bparees yeah I know I need to review those changes carefully because of
those footnotes. I was already tired yesterday and I wanted to create this
PR to let everyone know. Still thx for the early review :-)
On Jun 8, 2015 11:40 PM, "Alex Dellapenta" <notifications@github.com> wrote:

> Wowee, [image: :bow:].
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/openshift-docs/pull/467#issuecomment-110149346>
> .
>

soltysh commented:
I think I'm done, though there are a couple of places still remaining update to v1, but I just didn't know how to update them properly as of now. These places are:
```
$ ack "v1beta[13]"
dev_guide/deployments.adoc
298:            "version": "v1beta1",

admin_guide/selfprovisioned_projects.adoc
31:apiVersion: v1beta3
37:- apiVersion: v1beta1
48:- apiVersion: v1beta1

admin_guide/master_node_configuration.adoc
115:- v1beta3
172:  - v1beta3

architecture/core_objects/kubernetes_model.adoc
186:    "selfLink": "/api/v1beta1/replicationControllers",
188:    "apiVersion": "v1beta1",
194:            "selfLink": "/api/v1beta1/replicationControllers/docker-registry-1?namespace=default",
```
I'd like to ask authors of these files (@ironcladlou, @deads2k) to update those, don't forget about selfLinks :wink: 

@adellape @bparees @danmcp @mfojtik can I ask you guys for detailed review, I don't want to mess something up. I've made sure that each of those examples works so that part should be ok. But I might remove footnote links like previously @bparees noted or alike.

bparees commented on admin_guide/examples/geo-cache.json:
```diff
@@ -1,55 +1,74 @@
+        "selector": {
+          "ha-cache": "geo"
+        },
+        "portalIP": "",
```
you can just remove this.

bparees commented on admin_guide/examples/geo-cache.json:
```diff
@@ -1,55 +1,74 @@
+        "type": "ClusterIP",
+        "sessionAffinity": "None"
+      },
+      "status": {
```
delete?

bparees commented on admin_guide/examples/geo-cache.json:
```diff
@@ -1,55 +1,74 @@
         }
-    ]
+      },
+      "status": {
```
delete?

bparees commented on admin_guide/manage_nodes.adoc:
```diff
@@ -98,21 +98,26 @@ scheduled for the node.
+    }
   },
+  "spec": {
+    "externalID": "node3.example.com"
```
footnote this?

soltysh replied:
This is completely removed in the new docs.

bparees commented on architecture/core_objects/kubernetes_model.adoc:
```diff
@@ -243,35 +240,38 @@ attached:
+    "type": "ClusterIP",
+    "sessionAffinity": "None"
+  },
+  "status": {
```
delete

bparees commented on architecture/core_objects/openshift_model.adoc:
```diff
@@ -448,31 +438,33 @@ annotations.
+      "kubernetes"
+    ]
+  },
+  "status": {
```
delete?  (nothing we are illustrating as sample for a user to create from should have status i don't think.. we should only see status in samples of what is retrieved from the system)

bparees commented on creating_images/sti_testing.adoc:
```diff
@@ -143,45 +143,54 @@ a new STI builder image.
+  "spec": {
+    "triggers": [
+      {
+        "type": "github",
```
GitHub

bparees commented on creating_images/sti_testing.adoc:
```diff
@@ -143,45 +143,54 @@ a new STI builder image.
-        "exposeDockerSocket": true,
+        "from": {
+          "kind": "ImageStreamTag",
+          "name": "sti-image-builder:latest"
```
this won't work unless you also have them setup the imagestream.

or you can change the Kind to DockerImage and make the name be "openshift/sti-image-builder" again.

soltysh replied:
I think we want to advertise our ImageStreams here as much as possible, the example as is is working just fine and getting information that you're missing some IS is perfectly ok when someone tries to use this as an example, imho.

bparees replied:
in general I'd agree, on this one i'm less sure I agree.  we're not going to create an openshift/sti-image-builder stream out of the box, it's not really a normal thing to need/want.

soltysh replied:
Right..., I didn't take this into account. I'll update to use DockerImage.

bparees commented on dev_guide/builds.adoc:
```diff
@@ -366,13 +370,14 @@ trigger on:
+  "metadata": {
+    "name": "ruby-20-centos7"
+  },
+  "status": {
```
delete

bparees commented on dev_guide/builds.adoc:
```diff
@@ -366,13 +370,14 @@ trigger on:
+  },
+  "status": {
+    "dockerImageRepository": ""
+  }
```
this example is broken...  (it was broken even before you touched it).  this ImageStream doesn't actually point anywhere (no docker repository).  looks like it was supposed to point to ruby-20-centos7.

bparees replied:
ooops, nevermind. it's going to use the default internal registry and the metadata.name.

bparees commented on dev_guide/deployments.adoc:
```diff
@@ -34,59 +34,60 @@ new deployment every time a Docker image tag changes:
 ====
 
-[source,json]
+[source.json]
```
in other places you made the periods into commas, here you made a comma into a period.....?

bparees commented:
@soltysh comments made.

soltysh commented:
Rebased, @bparees addressed your comments with 2 exceptions (one about footnode for node - that part is gone in latest docs, two - replied inline). 
@adellape @bparees appreciate your final review and if you're ok I'd like to get it merged as this touches quite a lot and is conflicting very frequently ;)

soltysh commented:
@bparees updated.

adellape commented on architecture/core_objects/openshift_model.adoc:
```diff
@@ -67,84 +67,87 @@ OpenShift are immutable.
 
 Like Docker, images are retrieved within the scope of an image stream. The sample image below is from the `origin-ruby-sample` image stream, and can be retrieved by asking for the `ImageStreamImage` with name `origin-ruby-sample@sha256:4d3a646b58685449179a0c61ad4baa19a8df8ba668e0f0704b9ad16f5e16e642`.
 
+====
```
It looks like the opening example blocks being introduced throughout this topic are missing their closing `====`s, which is messing up any section titles that follow.

adellape commented on dev_guide/builds.adoc:
```diff
@@ -26,7 +26,9 @@ A buildConfig is a REST object, which can be used in a POST to the API server to
 every time a Docker image tag or the source code changes:
 
-[source.json]
+====
```
Similar issue in this topic w/ a few `====` blocks.

adellape commented:
@soltysh LGTM other than the handful of breaking example blocks.

soltysh commented:
@adellape fixed, hopefully all of them. Final review and merge :facepunch:

adellape commented:
@soltysh LGTM, thanks! :clap: Will merge after build passes.

soltysh commented:
:cocktail:
